packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataDetailsViewPageDescriptionProvider.java

```diff
@@ -0,0 +1,216 @@
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.RepresentationMetadata;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.services.api.IRepresentationMetadataUpdateService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to provide support for details view for RepresentationMetadata.
+ *
+ * @author Jerome Gout
+ */
+@Service
+public class RepresentationMetadataDetailsViewPageDescriptionProvider implements IPropertiesDescriptionRegistryConfigurer {
+    private final IIdentityService identityService;
+
+    private final ILabelService labelService;
+
+    private final IRepresentationMetadataUpdateService representationMetadataUpdateService;
+
+    public RepresentationMetadataDetailsViewPageDescriptionProvider(IIdentityService identityService, ILabelService labelService,
+            IRepresentationMetadataUpdateService representationMetadataUpdateService) {
+        this.identityService = Objects.requireNonNull(identityService);
+        this.labelService = Objects.requireNonNull(labelService);
+        this.representationMetadataUpdateService = Objects.requireNonNull(representationMetadataUpdateService);
+    }
+
+    @Override
+    public void addPropertiesDescriptions(IPropertiesDescriptionRegistry registry) {
+        registry.add(this.getPageDescription());
+    }
+
+    private PageDescription getPageDescription() {
+        String id = UUID.nameUUIDFromBytes("representation-metadata".getBytes()).toString();
+
+        List<AbstractControlDescription> controls = this.createControlDescriptions();
+        GroupDescription groupDescription = this.createGroupDescription(controls);
+
+        Predicate<VariableManager> canCreatePagePredicate = variableManager ->  variableManager.get(VariableManager.SELF, Object.class)
+                .filter(RepresentationMetadata.class::isInstance)
+                .isPresent();
+
+        return this.createPageDescription(id, groupDescription, canCreatePagePredicate);
+    }
+
+    private List<AbstractControlDescription> createControlDescriptions() {
+        BiFunction<RepresentationMetadata, String, IStatus> representationLabelWriter = (metadata, newLabel) -> {
+            // delegate the renaming of the representation to EditingContextEventProcessor
+            var representationId = metadata.getId();
+            Map<String, Object> parameters = new HashMap<>();
+            parameters.put(EditingContextEventProcessor.REPRESENTATION_ID, representationId.toString());
+            parameters.put(EditingContextEventProcessor.REPRESENTATION_LABEL, newLabel);
+            return new Success(ChangeKind.REPRESENTATION_TO_RENAME, parameters);
+        };
+
+        var labelControl = this.createLabelTextField(RepresentationMetadata::getLabel, representationLabelWriter);
```

Comment:
I don't understand the need for this complexity of providing a reader and writer to this function. It's used once in a very specific place, it can find out how to read /write a label by itself.